intera_interface/src/motion_interface/motion_waypoint_options.py

```diff
@@ -0,0 +1,207 @@
+                self._data.joint_tolerances.append(tol)
+        elif len(tol) == self._n_dim:
+            self._data.joint_tolerances = deepcopy(joint_tolerances)
+        else:
+            rospy.logerr('Invalid input to set_joint_tolerances!')
+
+    def set_max_linear_speed(self, speed = None):
+        if speed is None:
+            speed = 0.6; # default value
+        tol = clamp_float_warn(0.001, speed, float('inf'), 'max linear speed')
+        self._data.max_linear_speed = speed
+
+    def set_max_linear_accel(self, accel = None):
+        if accel is None:
+            accel = 0.6; # default value
+        tol = clamp_float_warn(0.001, accel, float('inf'), 'max linear accel')
+        self._data.max_linear_accel = accel
+
+    def set_max_rotational_speed(self, speed = None):
+        if speed is None:
+            speed = 1.57; # default value
+        tol = clamp_float_warn(0.001, speed, float('inf'), 'max rotational speed')
+        self._data.max_rotational_speed = speed
+
+    def set_max_rotational_accel(self, accel = None):
+        if accel is None:
+            accel = 1.57; # default value
+        tol = clamp_float_warn(0.001, accel, float('inf'), 'max rotational accel')
+        self._data.max_rotational_accel = accel
+
+    def set_max_joint_accel(self, max_joint_accel = []):
+        """
+        @param max_joint_accel:
+        --> None:  populate with vector of default values
+        --> []: set tolerance vector to empty
+        --> float:  scale the maximum acceleration for each joint by this
+        --> [float]:  copy all elements of max_joint_accel. Checks length.
+        --> str:  use a motion preset
+        """
+        if max_joint_accel is None:
+            self.set_max_joint_accel('fast')
+        elif not max_joint_accel:
+            self._data.max_joint_accel = []
+        elif isinstance(max_joint_accel, float):
+            max_acc = self.get_accel_preset('express')
+            max_joint_accel = clamp_float_warn(0.001, max_joint_accel, 1.0, 'joint accel')
+            for i in range(0, self._n_dim):
+                max_acc[i] = max_acc[i]*max_joint_accel
+            self.set_max_joint_accel(max_acc)
+        elif isinstance(max_joint_accel, str):
+            self.set_max_joint_accel(self.get_accel_preset(max_joint_accel))
+        else:  # Vector of doubles!
+            self._data.max_joint_accel = max_joint_accel
+            if not self.check_array_consistency():
+                self._data.max_joint_accel = []
+
+
+    def set_label(self, label="default"):
+        if isinstance(label, str):
+            self._data.label = deepcopy(label)
+        else:
+            rospy.logerr('Input must be a string!')
+
+    def check_array_consistency(self):
+        """
+        @return: true iff all arrays in the data structure are self consistent.
+        """
+        a = len(self._data.joint_tolerances)
+        b = len(self._data.max_joint_accel)
+        if a == 0 or b == 0:
```

Comment:
It will evaluate each of these in turn, so this could work:
```
if a == 0 or b == 0 or a == b:
  return True
else:
  rospy.logerr('Inconsistent array length in WaypointOptions!') 
  return False